src/NHibernate/Impl/SessionIdLoggingContext.cs

```diff
@@ -15,18 +15,38 @@ public class SessionIdLoggingContext : IDisposable
 #else
 		private const string LogicalCallContextVariableName = "__" + nameof(SessionIdLoggingContext) + "__";
 #endif
-		private readonly Guid? _oldSessonId;
-		private readonly bool _hasChanged;
-		
+		private readonly Guid? _oldSessionId;
+		private bool _hasChanged;
+
+		[Obsolete("Please use SessionIdLoggingContext.CreateOrNull instead.")]
 		public SessionIdLoggingContext(Guid id)
 		{
 			if (id == Guid.Empty) return;
-			_oldSessonId = SessionId;
-			if (id == _oldSessonId) return;
+			_oldSessionId = SessionId;
+			if (id == _oldSessionId) return;
 			_hasChanged = true;
 			SessionId = id;
 		}
 
+		private SessionIdLoggingContext(Guid newId, Guid? oldId)
+		{
+			SessionId = newId;
+			_oldSessionId = oldId;
+			_hasChanged = true;
+		}
+
+		public static IDisposable CreateOrNull(Guid id)
+		{
+			if (id == Guid.Empty)
+				return null;
```

Comment:
This shortcut will not play nice with cases where usages of sessions having an id are mixed with sessions not having one. That is likely an extreme corner case, though.

Reply:
Well this case was not supported anyway:
https://github.com/nhibernate/nhibernate-core/blob/1711b5c5162e9981fb6d83aaae8810b09aa97876/src/NHibernate/Impl/SessionIdLoggingContext.cs#L21-L28

As you can see `_hasChanged` is never set for sessions without `id`

Reply:
ok